### ISSUE TITLE
Design rewards for finality signatures

### ISSUE THREAD

**Issue by piotr-dziubecki** [casper-network/casper-node#3671]
What it says on the tin

**Comment by fizyk20** [comment]
Meeting summaries

2023-04-14

We walked through the diagram illustrating the lifecycle of a finality signature and through the corresponding code. We discussed the implications of that lifecycle and the components involved for the design of the rewards scheme and the complexity of the implementation. Based on our review, it seems that hooking into the finality signature lifecycle should be relatively simple, and that only a few components will have to be altered.

2023-04-21

We looked at the code of the auction contract as a refresher on how it works and where the rewards distribution happens. We outlined some general changes that will have to be made to the code. Alex presented an early draft of the design document.
